packages/protocol/test/layer1/based/TaikoL1TestBase.sol

```diff
@@ -255,14 +244,11 @@ abstract contract TaikoL1TestBase is TaikoTest {
         signature = abi.encodePacked(r, s, v);
     }
 
-    function giveEthAndTko(address to, uint256 amountTko, uint256 amountEth) internal {
-        vm.deal(to, amountEth);
-        tko.transfer(to, amountTko);
-
-        vm.prank(to, to);
-        tko.approve(address(L1), amountTko);
+    function giveEthAndDepositBond(address to, uint256 amountEth) internal {
+        vm.deal(to, amountEth + 1000 ether); // Extra eth compensates for block proposals
```

Comment:
this extra amount granted here breaks a bit the logic of the function, if I write tests and use it I would expect only the `amountEth` to be granted. Notice that if not I will need to keep it in mind for balance checks, and if you decide to change the number granted here all my tests will break.

Reply:
I agree. It wouldn't be fair to be done this way when writing from scratch, however, this is done here to keep it in line with the former tests of Taiko, where besides an arbitrary taiko token amount (10,000 TKO tokens), additional 1000 eth was sent to process block proposals: https://github.com/NethermindEth/taiko-mono-core-rollup/pull/2/files/4dfc2ec79c44101a0cc457b0fa23462b9028c7c4#diff-e92c7ce15871475760272c5000c1e390acb53eea8c383f38ddc8d7dd092f9a70L15

Reply:
can you keep it like the original function just for it to be more extensible and more friendly if future tests are needed. Kind of:
```solidity
function giveEthAndDepositBond(address to, uint256 bondEth, uint256 proposalEth) internal {
        vm.deal(to, amountEth + proposalEth);
```
That way you can control the amount in the tests

Reply:
Fair. Just pushed this change